DirectEmployers/MyJobs
# [PD-3881] [PD-3896] Reporting Fixes

## What's this PR do?
This PR covers a few issues in regards to reporting

1) Back/Forward navigation stopped working after we removed history.listen handling (https://github.com/DirectEmployers/MyJobs/pull/3615 ... yeah, I'm the one who merged it 😭  )
2) When on the Export view for a report, if you select "Export" on another report, the navigation does not work
3) The spinner didn't always show up when it needed to during navigation in Export/Preview.
4) The report list did not populate if the user was on Preview/Export and refreshed their browser.
## Where should the reviewer start?
Reporting.

## How should this be manually tested?
Apart from testing the exact cases mentioned above, check navigation in general. Ensure back, forward, reload and any other combination of navigation works properly and nothing is out of place.

## PR Checklist

### Front-End
